### PR TITLE
Add an extensions API endpoint for downloading signatures

### LINKED ISSUE #504: As a user I can view container image signatures
Author: @ipanova (ipanova@redhat.com)

Redmine Issue: 9513, https://pulp.plan.io/issues/9513

---

Provide an extension of the docker/distribution API. 

Assuming a signed image has already been pushed into the Pulp Container Registry, one can read the signature(s)( any user with  access to the image) using the following calls:

* `GET /extensions/v2/<namespace>/<name>/signatures/`
* `GET /extensions/v2/<namespace>/<name>/signatures/<imagesha256digest>`

See https://docs.openshift.com/container-platform/3.10/admin_guide/image_signatures.html#reading-image-signatures-via-registry-api and https://github.com/openshift/origin/pull/12504/files

### PR DESCRIPTION
**Pull request by lubosmj**
closes #504

### CODE REVIEW

File: pulp_container/app/registry_api.py [hunk 1]
```diff
@@ -970,3 +975,42 @@ def receive_artifact(self, chunk):
                 artifact = Artifact.objects.get(sha256=artifact.sha256)
                 artifact.touch()
             return artifact
+
+
+class Signatures(ContainerRegistryApiMixin, ViewSet):
+    """A ViewSet for image signatures."""
+
+    lookup_value_regex = "sha256:[0-9a-f]+"
```
Comment (mdellweg) [inline pulp_container/app/registry_api.py:983]: Let's be specific:
```suggestion
    lookup_value_regex = "sha256:[0-9a-f]{64}"
```

File: pulp_container/app/registry_api.py [hunk 2]
```diff
@@ -292,7 +292,12 @@ def default_response_headers(self):
         Provide common headers to all responses.
         """
         headers = super().default_response_headers
-        headers.update({"Docker-Distribution-Api-Version": "registry/2.0"})
+        headers.update(
+            {
+                "Docker-Distribution-Api-Version": "registry/2.0",
+                "X-Registry-Supports-Signatures": "1",
```
Comment (goosemania) [inline pulp_container/app/registry_api.py:298]: This header name is already defined in stages file https://github.com/pulp/pulp_container/blob/signing/pulp_container/app/tasks/sync_stages.py#L43, how about putting it into the constants and using it here and there?

File: pulp_container/app/registry_api.py [hunk 3]
```diff
@@ -970,3 +970,42 @@ def receive_artifact(self, chunk):
...
+
+        return Response(self.get_response_data(signatures))
+
+    @staticmethod
+    def get_response_data(signatures):
+        """Extract version, type, name, and content from the passed signature data."""
+        data = []
+        for signature in signatures:
+            signature = {
+                "schemaVersion": 2,
```
Comment (ipanova) [inline pulp_container/app/registry_api.py:1005]: that's not correct, version was correct https://github.com/containers/image/blob/b55fb86c28b7d743cf59701332cd78d4294c7c54/docker/docker_image_src.go#L559
https://github.com/containers/image/blob/f2c99a9b1a0cd6673aab0d538b48d68562702b04/docker/docker_client.go#L76

Comment (goosemania) [reply pulp_container/app/registry_api.py:1005]: Good find, "schemaVersion" worked much better!

@ipanova , https://github.com/containers/image/blob/f2c99a9b1a0cd6673aab0d538b48d68562702b04/docker/docker_client.go#L76 shows in the comment how json is being serialized, and that it's this sting and not the attribute name is used. But it is confusing, why not to make them the same :woman_shrugging:

Comment (ipanova) [reply pulp_container/app/registry_api.py:1005]: ok that's confusing as hell

Comment (lubosmj) [reply pulp_container/app/registry_api.py:1005]: `Version` stands for a parsed value. A raw JSON value is stored as "schemaVersion".

**Review by goosemania — APPROVED** [review]

**Pull request merged**